src/mainwindow.cpp

```diff
@@ -572,10 +572,23 @@ void MainWindow::onImageLoaded() {
 
     loadJob_ = nullptr; // the job object will be freed later automatically
 
-    ui.view->setAutoZoomFit(ui.actionZoomFit->isChecked());
-    if(ui.actionOriginalSize->isChecked()) {
-      ui.view->zoomOriginal();
+    // set image zoom
+    Application* app = static_cast<Application*>(qApp);
+    Settings& settings = app->settings();
+    
+    if (settings.forceZoomFit()) {
+      ui.actionZoomFit->setChecked(true);
+      ui.view->setAutoZoomFit(true);
+      ui.view->zoomFit();
     }
+    else {
+      // like in onImageLoaded()
+      ui.view->setAutoZoomFit(ui.actionZoomFit->isChecked());
+      if(ui.actionOriginalSize->isChecked()) {
+        ui.view->zoomOriginal();
+      }
+    }
+    
```

Comment:
`ImageView::zoomFit` is always called in `imageview.cpp` when an image is set. Since the image is set below these lines, we don't need to call it and the above code can be simplified as:

```c++
    if(static_cast<Application*>(qApp)->settings().forceZoomFit()) {
      ui.actionZoomFit->setChecked(true);
    }
    ui.view->setAutoZoomFit(ui.actionZoomFit->isChecked());
    if(ui.actionOriginalSize->isChecked()) {
      ui.view->zoomOriginal();
    }
```